packages/medusa/src/services/shipping-option.ts

```diff
@@ -409,6 +444,10 @@ class ShippingOptionService extends TransactionBaseService {
       const optionRepo = manager.getCustomRepository(this.optionRepository_)
       const option = optionRepo.create(data as DeepPartial<ShippingOption>)
 
+      await this.validateAndMutatePrice(option, {
```

Comment:
comment: Depending on the answer of the previous comment it need to be changed or not

Reply:
I am not sure, I understand what this comment means :)

Reply:
sorry 😂 I wanted to say that if we go for returning the option instead of mutating it in place, then this needs to be refactored to use the returned option instead